### PR TITLE
IGNITE-17393 Some code rolled back from jraft sources and redone in a less invasive way.

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-17393

### CODE REVIEW

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/DefaultJRaftServiceFactory.java [hunk 1]
```diff
@@ -53,4 +49,12 @@ public DefaultJRaftServiceFactory(LogStorageFactory factory) {
         Requires.requireTrue(!StringUtils.isBlank(uri), "Blank raft meta storage uri.");
         return new LocalRaftMetaStorage(uri, raftOptions);
     }
+
+    @Override public LogEntryCodecFactory createLogEntryCodecFactory() {
```
Comment (SammyVimes) [inline modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/DefaultJRaftServiceFactory.java:53]: Override on the same line

File: modules/raft/src/test/java/org/apache/ignite/raft/jraft/storage/impl/VolatileRaftMetaStorageTest.java [hunk 2]
```diff
@@ -17,19 +17,18 @@
...
-
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.nullValue;
 import static org.hamcrest.Matchers.sameInstance;
 import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
+import org.apache.ignite.raft.jraft.entity.PeerId;
```
Comment (SammyVimes) [inline modules/raft/src/test/java/org/apache/ignite/raft/jraft/storage/impl/VolatileRaftMetaStorageTest.java:27]: are you sure about import order?

File: modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/LogStorageFactory.java [hunk 3]
```diff
@@ -33,5 +34,5 @@ public interface LogStorageFactory extends AutoCloseable {
      * @param raftOptions Raft options.
      * @return Log storage.
      */
-    LogStorage getLogStorage(String groupId, RaftOptions raftOptions);
+    LogStorage createLogStorage(String groupId, RaftOptions raftOptions);
```
Comment (SammyVimes) [inline modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/LogStorageFactory.java:37]: groupId was uri AFAIR

**Review by SammyVimes — CHANGES_REQUESTED** [review]

File: modules/raft/src/test/java/org/apache/ignite/raft/jraft/storage/SnapshotExecutorTest.java [hunk 4]
```diff
@@ -16,20 +16,26 @@
  */
 package org.apache.ignite.raft.jraft.storage;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
[the remark this answers is not in the document]
Comment (ibessonov) [reply modules/raft/src/test/java/org/apache/ignite/raft/jraft/storage/SnapshotExecutorTest.java:19]: Thank you

File: modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/impl/IgniteJraftServiceFactory.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    /**
+     * Updates a value of {@link RaftMetaStorageFactory} in this service factory.
+     */
+    public IgniteJraftServiceFactory setRaftMetaStorageFactory(RaftMetaStorageFactory raftMetaStorageFactory) {
+        this.raftMetaStorageFactory = raftMetaStorageFactory;
+
+        return this;
+    }
+
+    @Override public LogStorage createLogStorage(final String groupId, final RaftOptions raftOptions) {
```
[the remark this answers is not in the document]
Comment (ibessonov) [reply modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/impl/IgniteJraftServiceFactory.java:68]: I'll fix it

File: modules/raft/src/test/java/org/apache/ignite/raft/jraft/storage/impl/VolatileRaftMetaStorageTest.java (hunk 2 above)
Comment (ibessonov) [reply modules/raft/src/test/java/org/apache/ignite/raft/jraft/storage/impl/VolatileRaftMetaStorageTest.java:27]: No

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/DefaultJRaftServiceFactory.java (hunk 1 above)
Comment (ibessonov) [reply modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/DefaultJRaftServiceFactory.java:53]: Not this time, I'm not gonna touch JRaft code.

**Review by SammyVimes — APPROVED** [review]
LGTM

**Pull request merged**